exporters/metric/stdout/stdout.go

```diff
@@ -130,26 +133,39 @@ func InstallNewPipeline(config Config) (*push.Controller, error) {
 
 // NewExportPipeline sets up a complete export pipeline with the recommended setup,
 // chaining a NewRawExporter into the recommended selectors and batchers.
-func NewExportPipeline(config Config, period time.Duration) (*push.Controller, error) {
+func NewExportPipeline(config Config, period time.Duration, options ...push.Option) (*push.Controller, error) {
 	selector := simple.NewWithExactMeasure()
 	exporter, err := NewRawExporter(config)
 	if err != nil {
 		return nil, err
 	}
 	batcher := ungrouped.New(selector, exporter.config.LabelEncoder, true)
-	pusher := push.New(batcher, exporter, period)
+	pusher := push.New(batcher, exporter, period, options...)
 	pusher.Start()
 
 	return pusher, nil
 }
 
-func (e *Exporter) Export(_ context.Context, checkpointSet export.CheckpointSet) error {
+func (e *Exporter) encodeResource(resource *resource.Resource) string {
```

Comment:
What happens if the exporter exports for two or more resources? As is the case when this exporter is registered with multiple controllers.

Reply:
I guess I was hoping no one would say this. :)

OTOH, that was my motivation for adding the comment about immutability and using pointers, so I'll add a map of resource to encoded value here.